Frameworks/Foundation/NSURLRequest.mm

```diff
@@ -208,4 +214,37 @@ - (void)dealloc {
     [super dealloc];
 }
 
+- (void)_lazyInitProperties {
+    @synchronized(self) {
```

Comment:
That requires you to store an additional once token... and then make sure that it doesn't lie when you set _properties manually (through copying or decoding)...